src/corelibs/U2View/src/ov_sequence/sequence_info/DNAStatisticsTask.cpp

```diff
@@ -368,14 +373,19 @@ void DNAStatisticsTask::computeStats() {
         return;
     }
 
+    // Max size to avoid cache overflow
+    static constexpr int MAX_SIZE = 1024 * 1024;
+    QByteArray meltTempSeq; 
     for (const U2Region& region : qAsConst(regions)) {
-        QList<U2Region> blocks = U2Region::split(region, 1024 * 1024);
+        QList<U2Region> blocks = U2Region::split(region, MAX_SIZE);
         for (const U2Region& block : qAsConst(blocks)) {
             if (isCanceled() || hasError()) {
                 break;
             }
             const QByteArray seqBlock = sequenceDbi->getSequenceData(seqRef.entityId, block, os);
             CHECK_OP(os, );
+
+            meltTempSeq.append(seqBlock, qMin(MAX_SIZE - meltTempSeq.size(), seqBlock.size()));
```

Comment:
Don't get your point.

> Return INVALID_TM if the sequence is above some reasonable limit (less than 1 block size).

What is "one block size"? If you open human_T1.fa and have no selection, the there is one block and its size is about 200'000 bases long, but if you select "1..10,20..30,40..50" there are three block, but just 30 bases long.

> This way you can move result.meltingTemp= out of the blocks loop.

 `result.meltingTemp=` is already out of block loop.

> result.meltingTemp  = sequenceSize > 10k ? INVALID : calculateTm()

`sequenceSize` could be as long as possible, but the selection size is several bases, these variables aren't related to each other.

Reply:
You changed the original algorithm and introduced a limit 1MB for the melting temperature to work.
I suggest to change 1MB (default buffer size) to a different constant that has a special meaning: MAX_LENGTH_TO_ESTIMATE_TEMPERATURE. I think the reasonable value here will be much less than 1MB.

Reply:
Geneious and SnapGene doesn't have this kind of limitation, I checked right now

Reply:
Bu now we do, right? I see that you compute TM only for the first 1Mb right?

Here is my logic: the old code didn't have this limitation. The new code has and set it to some non-related value. Could you please document this constant, why 1Mb?

Reply:
I can compute it for 1, 10 or 100 MB, doesn't matter. The point is the longest the sequence the less meling temperature changes.  I mean:
- If you have sequence 10 characters long its MT, 26.00 °C. If you add 1 character (length is 11 now) MT changes to 28.00 °C (2.00 °C difference).
- 100 characters long its MT, 75.81 °C. Add 10 characters - now MT 77.05 °C (1.24 °C difference).
- 1000 characters long its MT, 80.63 °C. Add 100 characters - now MT 80.43 °C (0.2 °C difference).
- 10000 characters long its MT, 80.46 °C. Add 1000 characters - now MT 80.49 °C (0.03 °C difference).

(this is the real data from human_T1.fa). You got the point, right? I take the sequence, add 10% of its length and compare MTs. The longer the sequence, the less MT differences we have. After some length this difference is so insignificantly, that there is no point even to calculate it. I decided to have a limit 1MB just because this is not a large amount of RAM and could be captured in one calculation process. The MT differences between 1MB and 10 MB would be so little, that there is no need of its calculation.

Also there is no need to write N\A because:
- Not a one app do this
- It's just weird and not clear what it means. We can calculate MT for the sequence of every length, the calculation algorithm has no upper sequence length limitation and allows us to do that, so why N\A?

Reply:
OK here is the question: the old had no limit, you have a limit and your limit comes from a very unrelated domain: buffer reader size.

A good solutions I see
1) Have no limit as it was before.
2) Put a reasonable limit that does not depend on buffer size

According to the data you put above the reasonable limit is 10k, right? 
So lets limit (and may be even speed up this way) TM calculation by 10k instead of using the current limit of 1Mb: which is read buffer size.

Reply:
Having no limit as it was before isn't a solution, because `Primer3` should have the whole sequence as an argument, it is just impossible to set it in parts - and if you had a 1GB sequence it may cause some complications. And I do not like 10k because this is much smaller than even sample sequences we have. For example, even our main sample file `human_T1` will change it's temperature, so an this length is too small. If you want to increase this number, let it be `200'000`.